src/cascadia/TerminalApp/Tab.cpp

```diff
@@ -300,7 +316,79 @@ void Tab::NavigateFocus(const winrt::TerminalApp::Direction& direction)
 // - <none>
 void Tab::ClosePane()
 {
-    _activePane->Close();
+    _rootPane->FindActivePane()->Close();
+}
+
+// Method Description:
+// - Setups all the event handlers we care about for a root pane. These are superset of
+//   the events that we register for every other pane in the tree, however, this method
+//   also calls _AttachEventHandlersToLeafPane and _AttachEventHandlersToControl, so there
+//   is no need to also call these on the root pane.
+// - It is called on initialization, and when the root pane changes (this is, it gets 
+//   splitted or collapsed after split).
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void Tab::_SetupRootPaneEventHandlers()
+{
+    if (const auto rootPaneAsLeaf = std::dynamic_pointer_cast<LeafPane>(_rootPane))
+    {
+        // Root pane also belongs to the pane tree, so attach the usual events, as for
+        // every other pane.
+        _AttachEventHandlersToLeafPane(rootPaneAsLeaf);
+        _AttachEventHandlersToControl(rootPaneAsLeaf->GetTerminalControl());
+
+        // When root pane closes, the tab also closes.
+        _rootPaneClosedToken = rootPaneAsLeaf->Closed([=](auto&& /*s*/, auto&& /*e*/) {

```

Comment:
Doesn't `[=]` mean we're making a copy of the entire object and operating on the copy instead of on the original?

This might need a dose of what #4051 is cooking up.

Reply:
No, it just captures `this` pointer by value. To operate on copy this would be `[*this]`.
As to events - yeah, you reminded me I was going to mention that in PR's description. Basically, events are being unsubscribed from in dtor, so they don't outlive the object and this is safe ('safe').

Reply:
> 'safe'

We should maybe leave a comment behind asserting why we thought it was "safe" so someone doesn't revise it later to make it unsafe or we can laugh at ourselves when some future crash proves that we didn't know as much about C++ as we all thought we did. :P

Reply:
Or better, decide on global pattern in such cases:

> All events that I introduced in LeafPane, ParentPane and Tab are being unsubscribed from in destructors of listener classes, therefore they can safely capture raw this pointer. But there may be other patterns: to always unsubscribe from events (which I've chosen), always capture strong/weak reference, or both (just for the feel of safety). Additionally, weak/strong capturing could also wait for #3999 and #3922. (Btw, if the event handler isn't removed, the listener object is destroyed and the listenee object is alive, then it's a leak, isn't it?)

I feel like this should be discussed together with https://github.com/microsoft/terminal/pull/4051#issuecomment-569844074 (which is concerned on what to do in case of option 2 and 3).

Reply:
Yeah probably.